ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_initiator_group.py

```diff
@@ -0,0 +1,379 @@
+    except Exception as ex:
+        return (False, False, f"Initiator group deletion failed | {ex}", {})
+
+
+def main():
+
+    fields = {
+        "state": {
+            "required": True,
+            "choices": ['present',
+                        'absent',
+                        'create'
+                        # 'append'
+                        ],
+            "type": "str"
+        },
+        "name": {
+            "required": True,
+            "type": "str",
+            "no_log": False
+        },
+        "description": {
+            "required": False,
+            "type": "str",
+            "no_log": False
+        },
+        "access_protocol": {
+            "choices": ['iscsi',
+                        'fc'
+                        ],
+            "required": False,
+            "type": "str",
+            "no_log": False
+        },
+        "host_type": {
+            "required": False,
+            "type": "str",
+            "no_log": False
+        },
+        "fc_tdz_ports": {
+            "required": False,
+            "type": "list",
+            "no_log": False,
+        },
+        "target_subnets": {
+            "required": False,
+            "type": "list",
+            "no_log": False,
+        },
+        "iscsi_initiators": {
+            "required": False,
+            "type": "list",
+            "no_log": False,
+        },
+        "fc_initiators": {
+            "required": False,
+            "type": "list",
+            "no_log": False,
+            "default": None
+        },
+        "app_uuid": {
+            "required": False,
+            "type": "str",
+            "no_log": False
+        },
+        "metadata": {
+            "required": False,
+            "type": "dict",
+            "default": None
+        }
+    }
+
+    default_fields = utils.basic_auth_arg_fields()
+    fields.update(default_fields)
+    required_if = [('state', 'create', ['access_protocol'])]
+    module = AnsibleModule(argument_spec=fields, required_if=required_if)
+    if client is None:
+        module.fail_json(msg='the python nimble-sdk module is required.')
+
+    hostname = module.params["hostname"]
+    username = module.params["username"]
+    password = module.params["password"]
+    state = module.params["state"]
+    initiator_group_name = module.params["name"]
+    description = module.params["description"]
+    access_protocol = module.params["access_protocol"]
+    host_type = module.params["host_type"]
+    fc_tdz_ports = module.params["fc_tdz_ports"]
+    target_subnets = module.params["target_subnets"]
+    iscsi_initiators = module.params["iscsi_initiators"]
+    fc_initiators = module.params["fc_initiators"]
+    app_uuid = module.params["app_uuid"]
+    metadata = module.params["metadata"]
+
+    if (username is None or password is None or hostname is None):
+        module.fail_json(
+            msg="Storage system IP or username or password or backend protocol is null.")
```

Comment:
Use something else but null. Rephrase as "Missing variables: hostname, username or password is mandatory."

Reply:
ok